Add Madgraph for HINPbPbSpring23 central campaign

stahlleiton opened the pull request:
This PR adds the HINPbPbSpring23 templates for MadGraph central production.

DickyChant commented:
So HIN L2s are happy (at least for MG5 ones).

On the technical side, we need special CMSSW version (`CMSSW_13_0_18_HeavyIon`, for latest nPDF) and corresponding `scram_arch` `el8_amd64_gcc11`, I can help on putting this in the script maybe? if we are all happy

sihyunjeon commented:
@stahlleiton hi thanks for doing all the templates. I believe you copy pasted them from existing ones in this repo?

What are the processes you need?

@DickyChant for the gridpack production i think we need
a. Genproductions to change to that cmssw branch that you mention (either in master or another separate branch). Machine picks up one genproductions branch and make gridpack from there
b. Change our imgae that we use in gridpack machine

a. I think we should ask GEN, b. I can take care

DickyChant commented:
> @stahlleiton hi thanks for doing all the templates. I believe you copy pasted them from existing ones in this repo?
> 
> What are the processes you need?
> 
> @DickyChant for the gridpack production i think we need a. Genproductions to change to that cmssw branch that you mention (either in master or another separate branch). Machine picks up one genproductions branch and make gridpack from there b. Change our image that we use in gridpack machine. We use slc7 now.
> 
> a. I think we should ask GEN, b. I can take care

for a. does it mean that the script won't accept those options?

stahlleiton commented:
I started from the Run3Summer23 campaign files and changed the beam energy and added the use of nuclear PDFs in the templates.

Regarding the genproduction branch, we have been using the master branch which works so far also for CMSSW_13_0_18_HeavyIons for MadGraph and Powheg.

For the process, I will copy the list in a follow up comment

sihyunjeon commented:
@DickyChant for now we choose a branch in genproductions so it directly takes the default cmssw from there. We can technically pass down which cmssw we want to use but this needs geovanny to make a clickable option in our web interface.

stahlleiton commented:
> @stahlleiton hi thanks for doing all the templates. I believe you copy pasted them from existing ones in this repo?

I started from Run3Summer23 campaign and changed the beam energy and changed the templates to use nuclear PDFs.

DickyChant commented:
> > @DickyChant for now we choose a branch in genproductions so it directly takes the default cmssw from there. We can technically pass down which cmssw we want to use but this needs geovanny to make a clickable option in our web interface.
> 
> If possible, I think it would be better if the CMSSW is not taken by default from the genproductions (since creating a new branch just to change the default CMSSW release would immediately make that branch go out of sync with the master branch)

I feel unfortunately the chance for the development is tiny however I just came up with two ugly solutions

1. We go with adding one more branch together with a CI that syncs this branch with master whenever there is a commit
2. We hard code the script to check the run card and see if heavy ion run bits are there (e.g 1: "Default_nPDFSETs" (I mean the handle that @stahlleiton made for picking up nuclear PDFs) and 2: the necessary bits for gamma_UPC run_card to set beam1 and beam2 number of protons/neutrons) we use the heavy ion release

stahlleiton commented:
> > > @DickyChant for now we choose a branch in genproductions so it directly takes the default cmssw from there. We can technically pass down which cmssw we want to use but this needs geovanny to make a clickable option in our web interface.
> > 
> > 
> > If possible, I think it would be better if the CMSSW is not taken by default from the genproductions (since creating a new branch just to change the default CMSSW release would immediately make that branch go out of sync with the master branch)
> 
> I feel unfortunately the chance for the development is tiny however I just came up with two ugly solutions
> 
>     1. We go with adding one more branch together with a CI that syncs this branch with master whenever there is a commit
> 
>     2. We hard code the script to check the run card and see if heavy ion run bits are there (e.g 1: "Default_nPDFSETs" (I mean the handle that @stahlleiton made for picking up nuclear PDFs) and 2: the necessary bits for gamma_UPC run_card to set beam1 and beam2 number of protons/neutrons) we use the heavy ion release

Can't we simply add an extra field in the json file "Campaigns/HINPbPbSpring23/HINPbPbSpring23.json" called "CMSSW" and use that instead?. So if present, use that otherwise, use the default from genproduction.

DickyChant commented:
> > > > @DickyChant for now we choose a branch in genproductions so it directly takes the default cmssw from there. We can technically pass down which cmssw we want to use but this needs geovanny to make a clickable option in our web interface.
> > > 
> > > 
> > > If possible, I think it would be better if the CMSSW is not taken by default from the genproductions (since creating a new branch just to change the default CMSSW release would immediately make that branch go out of sync with the master branch)
> > 
> > 
> > I feel unfortunately the chance for the development is tiny however I just came up with two ugly solutions
> > ```
> > 1. We go with adding one more branch together with a CI that syncs this branch with master whenever there is a commit
> > 
> > 2. We hard code the script to check the run card and see if heavy ion run bits are there (e.g 1: "Default_nPDFSETs" (I mean the handle that @stahlleiton made for picking up nuclear PDFs) and 2: the necessary bits for gamma_UPC run_card to set beam1 and beam2 number of protons/neutrons) we use the heavy ion release
> > ```
> 
> Can't we simply add an extra field in the json file "Campaigns/HINPbPbSpring23/HINPbPbSpring23.json" called "CMSSW" and use that instead (e.g. "CMSSW": "CMSSW_13_0_18_HeavyIon")?. So if present, use that otherwise, use the default from genproduction.

This should require development too, anyhow we can request a feature here: https://github.com/cms-PdmV/GridpackMachine

DickyChant commented:
> > > > > @DickyChant for now we choose a branch in genproductions so it directly takes the default cmssw from there. We can technically pass down which cmssw we want to use but this needs geovanny to make a clickable option in our web interface.
> > > > 
> > > > 
> > > > If possible, I think it would be better if the CMSSW is not taken by default from the genproductions (since creating a new branch just to change the default CMSSW release would immediately make that branch go out of sync with the master branch)
> > > 
> > > 
> > > I feel unfortunately the chance for the development is tiny however I just came up with two ugly solutions
> > > ```
> > > 1. We go with adding one more branch together with a CI that syncs this branch with master whenever there is a commit
> > > 
> > > 2. We hard code the script to check the run card and see if heavy ion run bits are there (e.g 1: "Default_nPDFSETs" (I mean the handle that @stahlleiton made for picking up nuclear PDFs) and 2: the necessary bits for gamma_UPC run_card to set beam1 and beam2 number of protons/neutrons) we use the heavy ion release
> > > ```
> > 
> > 
> > Can't we simply add an extra field in the json file "Campaigns/HINPbPbSpring23/HINPbPbSpring23.json" called "CMSSW" and use that instead (e.g. "CMSSW": "CMSSW_13_0_18_HeavyIon")?. So if present, use that otherwise, use the default from genproduction.
> 
> This should require development too, anyhow we can request a feature here: https://github.com/cms-PdmV/GridpackMachine

I would imagine this is a two line change actually? 

If we have that in the json fields we append it, otherwise no, for this line https://github.com/cms-PdmV/GridpackMachine/blob/7e34f08a9fd2aa94e190a5b33c7d613b342230ee/src/gridpack.py#L500

stahlleiton commented:
> What are the processes you need?

The process are:

- Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-50_amcatnloFXFX-pythia8
- Cards/MadGraph5_aMCatNLO/DY/DYto2L-2Jets_MLL-10to50_amcatnloFXFX-pythia8
- Cards/MadGraph5_aMCatNLO/W/WtoLNu-2Jets_amcatnloFXFX-pythia8
- Cards/MadGraph5_aMCatNLO/TT/TT-2Jets_amcatnloFXFX-pythia8  (added in this PR)

stahlleiton commented:
Any news on this?

DickyChant commented:
> Any news on this?

Thanks for reminder, in two weeks there will be a preMccM meeting and we will discuss there.

In my point of view we should have them if common background team doesn’t feel a huge burden 

1: normally heavy ion campaigns are always spring which means it will get used at a different time other than pp, so we could benefit from adding HIN requests to keep the cores busy
2: being saying so, the requests being added here would need to be taken care by common background team. I don’t want to have them being overloaded.


I also could understand if there is going to be blame on HIN not so common… Speaking of being common, if there is a low PU run extended from a pp ref, then the gridpacks here would be a good fit for those runs as well. For which @stahlleiton do we have some preliminary plan on that? Would be a nice point for supporting so.

sihyunjeon commented:
hey sorry for disappearing from this, somehow based on my private chats with sitian i misunderstood as if HIN wanted to separately do things themselves. my bad.

once we have rereco + 24 campaign set up, we can converge on having central production for HIN as well
